dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/processor/TaskDispatchProcessor.java

```diff
@@ -131,7 +132,7 @@ public void process(Channel channel, Command command) {
                 }
 
                 // check if the OS user exists
-                if (!OSUtils.getUserList().contains(taskExecutionContext.getTenantCode())) {
+                if (PropertyUtils.getBoolean(Constants.SUDO_ENABLE, true) && !OSUtils.getUserList().contains(taskExecutionContext.getTenantCode())) {
```

Comment:
I don't think we should add this sudo judgment here.
It's need to check tenant exist if we specify it in workflow, independent of whether or not you can sudo tenant.
WDYT?